src/pytesmo/utils.py

```diff
@@ -272,4 +275,134 @@ def array_dropna(*arrs):
 
     if len(arrs_dropna) == 1: arrs_dropna = arrs_dropna[0]
 
-    return tuple(arrs_dropna)
\ No newline at end of file
+    return tuple(arrs_dropna)
+
+def derive_edge_parameters(src, ref, perc_src, perc_ref):
+    '''
+    Method to compute the regression parameters and new percentile values for 
+    the edge matching in CDF matching, based on a linear scaling model.
+
+    Parameters
+    ----------
+    src : numpy.array
+        input dataset which will be scaled
+    ref : numpy.array
+        src will be scaled to this dataset
+    edge_src : list
+        list with low and high edges (in percentile values) of src
+    edge_ref : list
+        list with low and high edges (in percentile values) of ref
+
+    Returns
+    -------
+    parms_lo : tuple
+        slope and intercept parameters to scale the lower edge.
+    parms_hi : tuple
+        slope and intercept parameters to scale the higher edge.
+    perc_ref : list-like
+        new percentile values after regression
+    '''            
+    # select higher and lower edges
+    x_lo = src[src <= perc_src[1]] - perc_src[1]
+    y_lo = ref[ref <= perc_ref[1]] - perc_ref[1]
+    x_hi = src[src >= perc_src[-2]] - perc_src[-2]
+    y_hi = ref[ref >= perc_ref[-2]] - perc_ref[-2]
+    
+    # calculate least squares regression parameters
+    def return_regress(x,y,
+                       where,
+                       perc_src=perc_src,
+                       perc_ref=perc_ref):
+        l = min(len(x),len(y))
+        x, y = x[:l], y[:l]
```

Comment:
Could you rename `l` to `n`? `l` is sometimes hard to distinguish from `1`, depending on the used font. (See also https://www.python.org/dev/peps/pep-0008/#names-to-avoid)